ambry-cloud/src/main/java/com/github/ambry/cloud/azure/StorageClient.java

```diff
@@ -420,13 +420,19 @@ protected abstract BlobServiceAsyncClient buildBlobServiceAsyncClient(HttpClient
 
   /**
    * Handle any errors. For now, we update the error metrics.
-   * @param throwable associated with this error.
+   * @param T associated with this error.
    */
-  private void onError(Throwable throwable) {
-    logger.error("Blob store operation failed due to exception: " + throwable.toString());
-    azureMetrics.storageClientOperationExceptionCount.inc();
-    // All retries must have been completed internally by now.
-    azureMetrics.storageClientFailureAfterRetryCount.inc();
+  private void onError(Throwable T) {
+    if (T instanceof BlobStorageException
+        && ((BlobStorageException) T).getErrorCode() == BlobErrorCode.BLOB_ALREADY_EXISTS) {
+      // Common case first
+      logger.debug("Blob store operation failed due to exception: " + T);
+    } else {
+      logger.error("Blob store operation failed due to exception: " + T.toString());
```

Comment:
.toString() is redundant, but minor.

Reply:
yeah i noticed it too, strangely intellij says its redundant on L429 in the if-case but not in else-case.

Reply:
done